File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/resolution/ICandidateSource.java
package com.foundgine.core.semantic.resolution;

import com.foundgine.core.abstractions.*;
import java.util.List;

public interface ICandidateSource {
	List<IdentityCandidate> findByIdentity(EntityId entityType, String identityValue);

	List<IdentityCandidate> findByRelationship(RelationshipId relationshipId, String sourceIdentityValue);
}

File: src/csharp/Foundgine.Core/Semantic/Resolution/ICandidateSource.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Resolution;

/// <summary>
/// The boundary through which real data may participate in semantic
/// resolution. Implementations live outside the semantic layer.
/// </summary>
public interface ICandidateSource
{
    IReadOnlyList<IdentityCandidate> FindByIdentity(
        EntityId entityType,
        string identityValue);

    IReadOnlyList<IdentityCandidate> FindByRelationship(
        RelationshipId relationshipId,
        string sourceIdentityValue);
}